File: src/main/java/com/teamwizardry/librarianlib/api/gui/components/ComponentVoid.java
package com.teamwizardry.librarianlib.api.gui.components;

import com.teamwizardry.librarianlib.api.gui.GuiComponent;
import com.teamwizardry.librarianlib.math.Vec2;

public class ComponentVoid extends GuiComponent<ComponentVoid> {

	public ComponentVoid(int posX, int posY, int width, int height) {
		super(posX, posY, width, height);
	}
	
	public ComponentVoid(int posX, int posY) {
		super(posX, posY);
	}

	@Override
	public void drawComponent(Vec2 mousePos, float partialTicks) {
		// NOOP
	}

}

File: src/main/java/com/teamwizardry/librarianlib/client/gui/components/ComponentVoid.kt
package com.teamwizardry.librarianlib.client.gui.components

import com.teamwizardry.librarianlib.client.gui.GuiComponent
import com.teamwizardry.librarianlib.common.util.math.Vec2d

class ComponentVoid : GuiComponent<ComponentVoid> {

    constructor(posX: Int, posY: Int, width: Int, height: Int) : super(posX, posY, width, height)
    constructor(posX: Int, posY: Int) : super(posX, posY)

    override fun drawComponent(mousePos: Vec2d, partialTicks: Float) {
        //NO-OP
    }

}
